narwhal/primary/src/primary.rs

```diff
@@ -713,6 +748,21 @@ impl PrimaryReceiverHandler {
             DagError::HeaderRequiresQuorum(header.digest())
         );
 
+        // Verify any randomness signature present in the header.
```

Comment:
`sync_header_batches` is an async fn, which means calling it won't do anything unless I await it, in which case it will block until headers are fetched, right? Or what am I missing?